tests/test_url.py

```diff
@@ -315,24 +317,32 @@
 )
 
 
-def _test_safe_url_func(url, *, encoding=UNSET, output, func):
+def _test_safe_url_func(
+    url: StrOrBytes,
+    *,
+    encoding: Any = UNSET,
```

Comment:
`encoding` is actually either `str` or literally `UNSET`, but `UNSET = object()` and `typing.Literal` doesn't support literals of arbitrary types. It may be possible to refactor the code to make typing more simple.

Reply:
Moving this to the code.

Reply:
Using a sentinel value on a test file was probably overkill on my part, we can probably switch to `Optional[str]` and use `None` as sentinel.